### PR TITLE
auto make pr for xla pinned hash

### PR DESCRIPTION
**Pull request by clee2000**
see https://docs.google.com/document/d/1jRT-on-v8bcTCHD8aGFH7IuCDR7JL37GlP6Z2EYgmyk/edit#

add file with hash, add nightly job that makes a pr to update the hash, pr needs to be manually accepted (for now? maybe can get bot to approve it automatically later)

### CODE REVIEW

File: .github/workflows/update-xla-commit-hash.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: Update xla commit hash
+
+on:
+  schedule:
+    - cron: 0 0 * * *
```
Comment (janeyx99) [inline .github/workflows/update-xla-commit-hash.yml:5]: Should we make this not start at the 0th minute at the 0th hour as I assume lots of workflows do this.

Also include comment the cron "every night at ___"

File: .github/workflows/update-xla-commit-hash.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  workflow_dispatch:
+
+env:
+  NEW_BRANCH_NAME: clee2000/xla/${{ github.run_id }}-${{ github.run_number }}-${{ github.run_attempt }}
```
Comment (janeyx99) [inline .github/workflows/update-xla-commit-hash.yml:9]: Assuming the clee2000 will be replaced?

File: .github/workflows/update-xla-commit-hash.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+jobs:
+  update-xla-commit-hash:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+          submodules: false
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (janeyx99) [inline .github/workflows/update-xla-commit-hash.yml:23]: just curious--why do we need the token here?

**Review by janeyx99 — APPROVED** [review]
looks good, since the dev infra team is a CODEOWNER of .github, these PRs will add us as reviewers so we'll be notified.

File: .github/workflows/update-xla-commit-hash.yml (hunk 3 above)
Comment (clee2000) [reply .github/workflows/update-xla-commit-hash.yml]: apparently we don't, i just copied this from somewhere else and assumed it was needed lol

**Pull request closed without merging**